src/db.rs

```diff
@@ -1779,16 +1783,25 @@ fn convert_options(opts: &[(&str, &str)]) -> Result<Vec<(CString, CString)>, Err
         .collect()
 }
 
-fn convert_values(values: Vec<*mut c_char>, values_sizes: Vec<usize>) -> Vec<Vec<u8>> {
+fn convert_values(
+    values: Vec<*mut c_char>,
+    values_sizes: Vec<usize>,
+    errors: Vec<*mut c_char>,
+) -> Vec<Result<Option<Vec<u8>>, Error>> {
```

Comment:
Do you think that `Option<Vec>` would be better than a potentially empty `Vec` here?

Personally, I would use `Ok(value.unwrap_or(Vec::new()))` to simplify api a little bit.

Reply:
Without `Option` there's no way to distinguish between key -> empty string and key not in the database. This way `multi_get` is consistent with `get` that returns `Result<Option<Vec>>`.